psi4/extras.py

```diff
@@ -58,3 +59,77 @@ def exit_printing():
 
 def get_input_directory():
     return _input_dir_
+
+
+# Add-Ons
+def _CMake_to_Py_boolean(cmakevar):
+    if cmakevar.upper() in ["1", "ON", "YES", "TRUE", "Y"]:
+        return True
+    else:
+        return False
+
+
+def _psi4_which(command):
+    # environment is $PSIPATH:$PATH, less any None values
+    lenv = {'PATH': ':'.join([os.path.abspath(x) for x in os.environ.get('PSIPATH', '').split(':') if x != '']) +
+                    ':' + os.environ.get('PATH')}
+    lenv = {k: v for k, v in lenv.items() if v is not None}
+
+    try:
+        dashout = subprocess.Popen(command, stdout=subprocess.PIPE, env=lenv)
+    except OSError as e:
+        return False
+    else:
+        return True
+
+
+def _plugin_import(plug):
+    import sys
+    if sys.version_info >= (3, 4):
+        from importlib import util
+        plug_spec = util.find_spec(plug)
+    else:
+        import pkgutil
+        plug_spec = pkgutil.find_loader(plug)
+    if plug_spec is None:
+        return False
+    else:
+        return True
+
+
+_addons_ = {
+    "ambit": _CMake_to_Py_boolean("@ENABLE_ambit@"),
+    "chemps2": _CMake_to_Py_boolean("@ENABLE_CheMPS2@"),
+    "dkh": _CMake_to_Py_boolean("@ENABLE_dkh@"),
+    "libefp": _CMake_to_Py_boolean("@ENABLE_libefp@"),
+    "erd": _CMake_to_Py_boolean("@ENABLE_erd@"),
+    "gdma": _CMake_to_Py_boolean("@ENABLE_gdma@"),
+    "pcmsolver": _CMake_to_Py_boolean("@ENABLE_PCMSolver@"),
+    "simint": _CMake_to_Py_boolean("@ENABLE_simint@"),
+    "dftd3": _psi4_which("dftd3"),
+    "cfour": _psi4_which("xcfour"),
+    "mrcc": _psi4_which("dmrcc"),
+    "v2rdm_casscf": _plugin_import("v2rdm_casscf"),
+}
+
+def addons(request=None):
+    """Returns boolean of whether Add-On *request* is available to Psi4,
+    either compiled in or searchable in $PSIPATH:$PATH, as relevant. If
+    *request* not passed, returns list of available Add-Ons.
+
+    """
+    if request is None:
+        return sorted([k for k, v in _addons_.items() if v])
+    return _addons_[request.lower()]
```

Comment:
I think if `request is None` then this would return a list that cannot index `_addons_` or `lower`.